File: App/Dashboard/Dashboard.DataAccess/Models/Entities/OrderPayment.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore;

namespace Dashboard.DataAccess.Models.Entities;

[Table("order_payments")]
public partial class OrderPayment
{
    [Key]
    [Column("id")]
    public long Id { get; set; }

    [Column("order_id")]
    public long OrderId { get; set; }

    [Column("payment_method_id")]
    public long PaymentMethodId { get; set; }

    [Column("payment_status_id")]
    public long PaymentStatusId { get; set; }

    [Column("amount", TypeName = "decimal(18, 2)")]
    public decimal Amount { get; set; }

    [Column("payment_date")]
    [Precision(6)]
    public DateTime? PaymentDate { get; set; }

    [Column("transaction_id")]
    [StringLength(100)]
    [Unicode(false)]
    public string? TransactionId { get; set; }

    [Column("notes")]
    [StringLength(255)]
    [Unicode(false)]
    public string? Notes { get; set; }

    [Column("created_at")]
    [Precision(6)]
    public DateTime CreatedAt { get; set; }

    [Column("last_modified")]
    [Precision(6)]
    public DateTime LastModified { get; set; }

    [ForeignKey("OrderId")]
    [InverseProperty("OrderPayments")]
    public virtual Order Order { get; set; } = null!;

    [ForeignKey("PaymentMethodId")]
    [InverseProperty("OrderPayments")]
    public virtual PaymentMethod PaymentMethod { get; set; } = null!;

    [ForeignKey("PaymentStatusId")]
    [InverseProperty("OrderPayments")]
    public virtual PaymentStatus PaymentStatus { get; set; } = null!;
}

File: App/backEnd/src/main/java/com/project/webbanhang/models/orders/OrderPayment.java
package com.project.webbanhang.models.orders;

import com.project.webbanhang.models.BaseEntity;
import com.project.webbanhang.models.PaymentMethod;
import com.project.webbanhang.models.PaymentStatus;
import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;

@Entity
@AllArgsConstructor
@NoArgsConstructor
@Setter
@Getter
@Builder
@Table(name = "order_payments")
public class OrderPayment extends BaseEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "order_id")
    private Order order;

    @ManyToOne
    @JoinColumn(name = "payment_method_id")
    private PaymentMethod paymentMethod;

    @ManyToOne
    @JoinColumn(name = "payment_status_id")
    private PaymentStatus paymentStatus;

    @Column(name = "amount")
    private Long amount;

    @Column(name = "payment_date")
    private LocalDateTime paymentDate;

    @Column(name = "transaction_id", nullable = true)
    private String transactionId;

    @Column(name = "notes")
    private String notes;
}
